src/sentry/ingest/ingest_consumer.py

```diff
@@ -384,9 +385,29 @@ def process_individual_attachment(message, projects) -> None:
         logger.exception("invalid individual attachment type: %s", attachment.type)
         return
 
+    new_attachment = attachment
+    # We need the event to get the proguard file
+    # TODO: What if we don't have the event? Do we drop the attachment?
```

Comment:
Is that considered a valid envelope tho? IIRC, attachments should be always bounded to an event (maybe there are exceptions for different types of attachments), so the envelope must contain the `eventId`.

Reply:
It does contain the event ID, but if the attachment arrives before the event is processed then I'm unable to get the event data to tell me which proguard file to use for the deobfuscation and that seems to be consistently the case in my local env